### DISCUSSION TITLE
Editing a component in Tina cms

### DISCUSSION THREAD

**Discussion by Kosiso29** [tinacms/tinacms#3765] [Help] [answered]
Please I'm struggling with editing a nextjs component in tinacms. I've only been able to edit blog pages, and the cli installation doesn't come out of the box with a demo editable react component. It's just demo blog pages. Please any help or demo on how to edit components?

**Comment by logan-anderson** [comment, marked as answer]
@Kosiso29  Have you read the docs on [contextual editing in react]( https://tina.io/docs/contextual-editing/react/) ? They should be able to get you started. 

We also have a couple of example sites if you would rather just see working code.
- Basic starter: https://github.com/tinacms/tina-barebones-starter
- Starter with more complex components and tailwind: https://github.com/tinacms/tina-cloud-starter


If you are still suck, do you mind providing a bit more info such as an example repo and what react component you are trying to edit.

Thanks for your question and please reach out if you are still stuck.

**Reply by Kosiso29** [reply]
@logan-anderson thanks for the quick reply. I've gone through the [contextual editing in react](https://tina.io/docs/contextual-editing/react/). I actually tried the tina-barebones-starter and it worked fine and I was able to edit things the way I want, but when I wanted to do the same thing in my existing nextjs app (after installing using `npx @tinacms/cli@latest init`) I discovered that I get the error 

> _tina_generated_client__WEBPACK_IMPORTED_MODULE_2__.client.queries.page is not a function

But the same call `client.queries.page` was made in the tina-barebones-starter without any errors.

**Reply by Kosiso29** [reply]
No worries, I've figured it out. Apparently I didn't define any collection named page in the config schema. Once I defined it, it started working. Thanks a lot for your help. I believe I'm good to go now.
